File: app/src/main/java/com/twismart/wallpapershd/ui/main/MainPagerAdapter.java
package com.twismart.wallpapershd.ui.main;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.util.Log;

import com.twismart.wallpapershd.R;
import com.twismart.wallpapershd.ui.main.ListWallpapersFragment;
import com.twismart.wallpapershd.utils.Constants;

/**
 * Created by sneyd on 12/3/2016.
* */

public class MainPagerAdapter extends FragmentPagerAdapter {

    private int[] titleFragments = {R.string.title_tab_all, R.string.title_tab_most_popular, R.string.title_tab_my_favorites};
    private Context context;

    public MainPagerAdapter(FragmentManager fragmentManager, Context context){
        super(fragmentManager);
        this.context = context;
    }

    @Override
    public int getCount() {
        return titleFragments.length;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return ListWallpapersFragment.newInstance(Constants.TypeListWallpapers.ALL.value);
            case 1:
                return ListWallpapersFragment.newInstance(Constants.TypeListWallpapers.MOST_POPULAR.value);
            default:
                return ListWallpapersFragment.newInstance(Constants.TypeListWallpapers.MY_FAVORITES.value);
        }
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return context.getString(titleFragments[position]);
    }
}

File: app/src/main/kotlin/com/twismart/wallpapershd/ui/main/MainPagerAdapter.kt
/*
 * Copyright (C) 2017 Sneyder Angulo.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.twismart.wallpapershd.ui.main


import android.app.Fragment
import android.app.FragmentManager
import android.content.Context
import android.support.v13.app.FragmentPagerAdapter
import com.twismart.wallpapershd.R
import com.twismart.wallpapershd.utils.Constants

class MainPagerAdapter(fragmentManager: FragmentManager, private val context: Context) : FragmentPagerAdapter(fragmentManager) {

    private val titleFragments: IntArray by lazy { intArrayOf(R.string.title_tab_all, R.string.title_tab_most_popular, R.string.title_tab_my_favorites) }

    override fun getCount() = titleFragments.size

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> WallpaperListFragment.newInstance(Constants.TypeListWallpapers.ALL.value)
            1 -> WallpaperListFragment.newInstance(Constants.TypeListWallpapers.MOST_POPULAR.value)
            else -> WallpaperListFragment.newInstance(Constants.TypeListWallpapers.MY_FAVORITES.value)
        }
    }

    override fun getPageTitle(position: Int): String = context.getString(titleFragments[position])
}